addons/core/addon/services/confirm.js

```diff
@@ -17,13 +17,21 @@ import config from 'ember-get-config';
 export default class ConfirmationsService extends Service {
   // =attributes
 
+  #enabled;
+
   /**
    * When the confirm service is disabled, it always returns resolving promises.
    * By default the service is enabled but it is useful to disable it for test
    * purposes.
    * @type {boolean}
    */
-  enabled = config.enableConfirmService ?? true;
+  get enabled() {
+    const config = getOwner(this).resolveRegistration('config:environment');
+    return this.#enabled ?? config.enableConfirmService ?? true;
```

Comment:
This return is a bit weird to me :sweat_smile: is something ember related or vanila js?

Reply:
We sometimes set these values in tests, which we need to account for here.  Essentially it's saying:  return the first value you find in the list:  private enabled, config enabled, or literally true.